src/rely/matchers/EqualsMatcher.re

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */;
+open CommonOption.Infix;
+open MatcherTypes;
+open EqualityValidator;
+
+type equalsFn('a) = ('a, 'a) => bool;
+
+type equalsMatcher('a) = (~equals: equalsFn('a)=?, 'a, 'a) => unit;
+
+let passMessageThunk = () => "";
+let defaultEqualityFn = (==);
+
+let suggestCustomEqualityForFloat = "Note that you are testing for float equality using (==). You probably need to specify a ~equals function to the matcher.";
```

Comment:
nit: should probably split string up and concatenate pieces so it fits in under 80 columns

(I don't care if you actually make this change if it will be annoying to rebase/whatever though)